src/modm/platform/dma/stm32/dma.hpp.in

```diff
@@ -38,103 +29,214 @@ namespace platform
+					memoryIncrement, peripheralIncrement, priority, circular);
+		}
+
+		static void
+		start()
+		{
+			Hal::clearInterruptFlags(uint32_t(Interrupt::Global) << (uint32_t(ChannelID) * 4));
+			ChannelHal::start();
+		}
+		static void
+		stop()
+		{
+			ChannelHal::stop();
+		}
+		
+		static DataTransferDirection
+		getDataTransferDirection()
+		{
+			return ChannelHal::getDataTransferDirection();
+		}
+		
+		static void
+		setAddresses(uint32_t sourceAddress, uint32_t destinationAddress)
+		{
+			ChannelHal::setAddresses(sourceAddress, destinationAddress);
+		}
+		static void
+		setSourceAddress(uint32_t address)
+		{
+			ChannelHal::setSourceAddress(address);
+		}
+		static void
+		setDestinationAddress(uint32_t address)
+		{
+			ChannelHal::setDestinationAddress(address);
+		}
+
+		static void
+		setMemoryIncrementMode(bool increment)
+		{
+			ChannelHal::setMemoryIncrementMode(increment);
+		}
+		static void
+		setPeripheralIncrementMode(bool increment)
+		{
+			ChannelHal::setMemoryIncrementMode(increment);
+		}
+
+		static void
+		setDataLength(std::size_t length)
+		{
+			ChannelHal::setDataLength(length);
+		}
+
+		static void
+		setTransferErrorIrqHandler(IrqHandler irqHandler)
+		{
+			transferError = irqHandler;
+		}
+		static void
+		setTransferCompleteIrqHandler(IrqHandler irqHandler)
+		{
+			transferComplete = irqHandler;
+		}
+		template <DmaBase::Request dmaRequest>
+		static void
+		setPeripheralRequest()
+		{
+			DMA_Request_TypeDef *DMA_REQ = reinterpret_cast<DMA_Request_TypeDef *>(Hal::DMA_CSEL);
+			DMA_REQ->CSELR &= ~(0x07 << (uint32_t(ChannelID) * 4));
+			DMA_REQ->CSELR |= uint32_t(dmaRequest) << (uint32_t(ChannelID) * 4);
+		}
+
+		static void
+		interruptHandler()
+		{
+			static const uint32_t TC_Flag { uint32_t(Interrupt::TransferComplete) << (uint32_t(ChannelID) * 4) };
+			static const uint32_t TE_Flag { uint32_t(Interrupt::Error) << (uint32_t(ChannelID) * 4) };
+
+			auto isr { Hal::getInterruptFlags() };
+			if (isr & TE_Flag) {
+				disable();
+				if (transferError)
+					transferError();
+			}
+			if ((isr & TC_Flag) and transferComplete)
+				transferComplete();
+
+			Hal::clearInterruptFlags(uint32_t(Interrupt::Global) << (uint32_t(ChannelID) * 4));
+		}
+
+		static void
+		enableInterruptVector(uint32_t priority = 1)
+		{
+			NVIC_SetPriority(DmaBase::Nvic<ID>::DmaIrqs[uint32_t(ChannelID)], priority);
+			NVIC_EnableIRQ(DmaBase::Nvic<ID>::DmaIrqs[uint32_t(ChannelID)]);
+		}
+		static void
+		disableInterruptVector()
+		{
+			NVIC_DisableIRQ(DmaBase::Nvic<ID>::DmaIrqs[uint32_t(ChannelID)]);
+		}
+
+		static void
+		enableInterrupt(Interrupt_t irq)
+		{
+			ChannelHal::enableInterrupt(irq);
+		}
+		static void
+		disableInterrupt(Interrupt_t irq)
+		{
+			ChannelHal::disableInterrupt(irq); 
+		}
+
+		template <Peripheral peripheral, Signal signal = Signal::NoSignal>
+		struct RequestMapping {
+		};
+
+	private:
+		static DmaBase::IrqHandler transferError;
+		static DmaBase::IrqHandler transferComplete;
```

Comment:
You can [use `static inline` now](https://en.cppreference.com/w/cpp/language/inline) to avoid the annoying instantiation of static member data in templates.

Reply:
Done.